### PR TITLE
Initial Project Setup with Core Components, NativeWind, Expo Router, and Login Screen

### PR DESCRIPTION
**Pull request by Ghass3n**
**Description**

This PR introduces the foundational setup for the application, including essential UI components, styling configurations, routing, and the initial login screen. The changes establish a structured architecture to ensure scalability and maintainability moving forward.

**Related Issue**

N/A

**Motivation and Context**

- The project required a structured setup with core UI components to be reusable across the app.
- NativeWind/Tailwind CSS was implemented to enhance styling efficiency and maintain a clean design.
- Expo Router was set up to manage navigation and improve project organization.
- Cleaned up unnecessary files to streamline the project structure.
- Initiated the login screen development as part of the authentication flow.

**How Has This Been Tested?**

- Tested in a local development environment to ensure proper UI rendering and navigation.
- Verified that all UI components function correctly within the login screen.
- Checked Expo Router navigation to confirm smooth screen transitions.
- Ensured Tailwind CSS styling is applied correctly and responsive.

**Visual Aids**

WhatsApp Image 2025-02-06 at 14 52 50_43448c24
image

**Types of Changes**

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ *]  New feature (non-breaking change which adds functionality)
- [ *]  Breaking change (fix or feature that would cause existing functionality to change)
- [ ]  Documentation Update (changes to the documentation)

**Checklist:**

- [ *]  My code follows the code style of this project.
- [ *]  I have performed a self-review of my own code.
- [ *]  I have commented my code where necessary, especially in hard-to-understand areas.
- [ *]  I have made corresponding changes to the documentation, if required.
- [ *]  I have added tests that prove my fix is effective or that my feature works as intended.
- [ *]  New and existing unit tests pass locally with my changes.
- [ *]  Any dependent changes have been merged and published in downstream modules.
- [ *]  I have read the [CONTRIBUTING](https://github.com/Ghass3n/KorporFront/compare/CONTRIBUTING.md) document.

### CODE REVIEW

**Comment by ahmedjaziri31** [comment]
check out the architecture here:
https://drive.google.com/file/d/1XeuZR6xF6vyf5gn0TZpVnx2WyaiIym32/view?usp=sharing

To make a refacto for more centralised project

File: components/ui/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import DividerWithText from "./dividerWithText";
+import Input from "./textInput";
+import PasswordInput from "./passwordInput";
+import RememberMeCheckbox from "./rememberMe";
+import PressableText from "./pressableText";
+
+
+
+export { OutlinedButton, SolidButton, LoginCard, GoogleButton, DividerWithText, Input, PasswordInput, RememberMeCheckbox, PressableText};
+
```
Comment (ahmedjaziri31) [inline components/ui/index.tsx]: important to do the same thing here in each part of the project that needs to be centralized
for example check my last shared link to a similar project

**Review by ahmedjaziri31 — CHANGES_REQUESTED** [review]

**Comment by Ghass3n** [comment]
**Description**  
This pull request introduces several enhancements and fixes to improve the authentication and signup process. The key changes include:  

- **Added the signup screen** to allow new users to register.  
- **Implemented date input logic** to handle user-provided date inputs efficiently.  
- **Added phone number input logic** with validation for better user experience.  
- **Enhanced the authentication process** to improve security and reliability.  
- **Refactored the project structure** to improve maintainability and organization.  

These changes improve the overall user experience and maintainability of the codebase.  

---

 **Related Issue**  

there is an issue with the app building process with the NDK version

---

#**Motivation and Context**  
The changes in this PR were motivated by the need to:  

- Provide a **fully functional signup screen** for user registration.  
- Ensure **correct and user-friendly input handling** for date and phone number fields.  
- Improve **security and efficiency** in the authentication process.  
- **Organize the project structure** to enhance readability and maintainability.  

By implementing these changes, we aim to create a smoother and more reliable user experience.  

---

 **How Has This Been Tested?**  
This update has been tested using the following methods:  

- **Local environment testing** to verify functionality.  
- **Manual testing** to ensure the signup screen works correctly.  
- **Unit tests** for input validation (date and phone number).  
- **Integration tests** to confirm authentication flow improvements.  
- **Edge case scenarios** tested for input fields (e.g., invalid dates, incorrect phone numbers).  

All tests passed successfully, ensuring these changes function as expected.  

---

 **Types of Changes**  
Please mark the type of change your PR introduces:  

- [x] **Bug fix** (non-breaking change which fixes an issue)  
- [x] **New feature** (non-breaking change which adds functionality)  
- [ ] **Breaking change** (fix or feature that would cause existing functionality to change)  
- [ ] **Documentation Update** (changes to the documentation)  

---

 **Checklist:**  
Before submitting this PR, please ensure that:  

- [x] My code follows the code style of this project.  
- [x] I have performed a self-review of my own code.  
- [x] I have commented my code where necessary, especially in hard-to-understand areas.  
- [x] I have made corresponding changes to the documentation, if required.  
- [x] I have added tests that prove my fix is effective or that my feature works as intended.  
- [x] New and existing unit tests pass locally with my changes.  
- [x] Any dependent changes have been merged and published in downstream modules.  
- [x] I have read the [[CONTRIBUTING](https://chatgpt.com/CONTRIBUTING.md)](./CONTRIBUTING.md) document.  

.

**Review by ahmedjaziri31 — APPROVED** [review]

File: components/ui/passwordInput.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        placeholder={placeholder}
+        placeholderTextColor="#A0A0A0"
+        secureTextEntry={!isPasswordVisible} // Toggle password visibility
+      />
+
+      {/* Eye Icon Button */}
+      <TouchableOpacity
+        onPress={() => setIsPasswordVisible(!isPasswordVisible)}
+        className="absolute right-1 transform -translate-y-1/2 flex items-center justify-center"
+        style={{ height: 48, width: 40 }} // Adjusted touchable area
```
Comment (ahmedjaziri31) [inline components/ui/passwordInput.tsx:25]: hak e tailwind

**Review by ahmedjaziri31 — APPROVED** [review]
